crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -85,6 +85,15 @@ def services
     end
   end
 
+  api :POST, "/api/upgrade/nodes", "Initiate the upgrade of all nodes"
+  api_version "2.0"
+  # This is gonna initiate the upgrade of all nodes.
+  # The method runs asynchronously, so there's a need to poll for the status and possible errors
+  def nodes
+    # FIXME: implement this as asynchronous call!
+    render json: Api::Upgrade.nodes
```

Comment:
I wouldn't render anything here, as I just realized it is a POST request. I'd do more sth like

```ruby
if Api::Upgrade.nodes
  head :ok
else
  render json: { error: "Node Upgrade failed" }, status :unprocessable_entity
end
```

we shouldn't expect the request to return anything else than a http code in case of success, and an error + http code in case of failure